flink-end-to-end-tests/flink-end-to-end-tests-common/src/main/java/org/apache/flink/tests/util/AutoClosableProcess.java

```diff
@@ -115,6 +115,22 @@ public void runBlocking(final Duration timeout) throws IOException {
 			}
 		}
 
+		public void runBlockingWithRetry(final int maxRetries, final Duration timeoutOfTry) throws IOException {
+			int retries = 0;
+			while (true) {
+				try {
+					runBlocking(timeoutOfTry);
+					break;
+				} catch (Throwable e) {
```

Comment:
We should only catch exceptions; throwables here would indicate some significant problem with the code for which we just want to fail right away.

Reply:
OK